tests/baselines/reference/objectSpread.types

```diff
@@ -571,8 +571,8 @@ let shortCutted: { a: number, b: string } = { ...o, a }
 
 // non primitive
 let spreadNonPrimitive = { ...<object>{}};
->spreadNonPrimitive : object
->{ ...<object>{}} : object
+>spreadNonPrimitive : {}

```

Comment:
The result of a spread is definitely an `object`, so the old types are a bit more accurate.

Reply:
Very true. The only observable effect I can think of is this, though:

```ts
let nonprim = { ...<object>{}};
nonprim = 12;
```

Reply:
OK, I was going to try to find an example, but apparently this works:
```ts
function huh(x: {}): object {
	return x;
}
```
Huh? Aren't these different types?

Reply:
I think this is intentional, and a consequence of the rule that "all object types are assignable to `object`". This would be fine, except that we (incorrectly) allow `number` to be assignable to `{}` on the grounds that it's like assigning `Number => {}`.

@DanielRosenwasser do you remember if this is correct?

Reply:
@andy-ms type `object` was introduced by TS@2.2 (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html). The example in the documentation seems to indicate that `object` is more accurate that `{}` when referring methods from **Object**. I used to find this [article](https://blog.mariusschulz.com/2017/02/24/typescript-2-2-the-object-type) very helpful for me to understand the differences. :)

Reply:
@ZheyangSong I think that was @andy-ms’ original point. 

I realized that you can get the former type with a special case in getSpreadType when left is {} and right is object. But this only applies for exactly the example above, which should only show up in functions named cloneRecord.

Reply:
I see. Thanks. :)